### PR TITLE
Add flex, grid and size

### PR DESCRIPTION
**Pull request by saralohr**
+ Updates comment style so comments aren't output in CSS
+ Make spacing more consistent in `config.scss`
+ Adds `Grid` component, `size` utilities, and `flex` utilities borrowed from SUITCSS with a few exclusions, and Sass-ified.

Documented in `ui-patterns`: https://github.com/teamsnap/ui-patterns/pull/7

### CODE REVIEW

File: src/base/config.scss [hunk 1]
```diff
@@ -3,11 +3,10 @@
 // TODO Revisit typographic scale to find common ground with marketing
 // TODO take audit of the museo fonts used and clean up heading font variables
 
-/**
- * State Color Settings
- *
- * These should stay consistent across themes
- */
+
+// * State Color Settings
```
Comment (williamdingwall) [inline src/base/config.scss]: I like seeing this

File: src/utils/size.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// * Percentage based size classes for a 24 column grid
+
+$cols: 24;
+
+@for $i from 1 through ($cols) {
```
Comment (williamdingwall) [inline src/utils/size.scss]: I like this approach for defining the widths

**Comment by saralohr** [comment]
> I wonder if it'd be a good idea to start adding responsive utilities too. Just turned off now.

I was wondering this as well. By turned off do you mean commented out? 

Sizing is probably the main one that we would need responsive utilities for, and everything else we could add on an as-necessary basis.

**Comment by williamdingwall** [comment]
Yeah I would mostly agree with that. For turning off, I was thinking something more along these lines...
```
$responsive: false !default;

$screen-xs:  480px !default;
$screen-sm:  768px !default;
$screen-md:  992px !default;
$screen-lg: 1200px !default;

$breakpoints: (
  'xs'      'screen and (min-width: #{$screen-xs})',
  'sm'     'screen and (min-width: #{$screen-sm})',
  'md'     'screen and (min-width: #{$screen-md})',
  'lg'	      'screen and (min-width: #{$screen-lg})'
);

/* Loop through widths .u-size#{$i}of#{$cols} to create base widths */

@if $responsive == true {
  /* Loop through breakpoints variable and widths .u-size#{$i}of#{$cols}--#{$alias} to create responsive widths */
}
```

that way if we did want to build in responsive versions of components, or add in container width variables we could base it all off a single $responsive variable that can be turned off or on per project.

**Comment by saralohr** [comment]
I like that idea a lot @williamdingwall. If it's cool with you I'll add that as a new ticket and separate PR.

**Pull request merged**